From 97edcb986d312ca1df2f93898a613810cfb17bd6 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Tue, 24 Oct 2023 13:25:42 +0200
Subject: [PATCH] ViewStateController.qml: Fix TypeError

Fixes:

Oct 24 09:46:03 qemux86-64 surface-manager[614]: [] [pmlog] surface-manager LSM {} (null), file:///usr/lib/qml/WebOSCompositorBase/controllers/ViewStateController.qml:128: TypeError: Cannot read property 'start' of undefined

Upstream-Status: Pending

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 .../controllers/ViewStateController.qml                    | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/base/qml/WebOSCompositorBase/controllers/ViewStateController.qml b/base/qml/WebOSCompositorBase/controllers/ViewStateController.qml
index 0963cbb..de48dca 100644
--- a/base/qml/WebOSCompositorBase/controllers/ViewStateController.qml
+++ b/base/qml/WebOSCompositorBase/controllers/ViewStateController.qml
@@ -124,8 +124,11 @@ Item {
             if (compositorWindow.displayId > 0)
                 return;
             if (event === "splash") {
-                if (appId != null && (showSplash || showSpinner))
-                    views.spinner.start(appId);
+                if (appId != null && (showSplash || showSpinner)) {
+                    if (views.spinner) {
+                        views.spinner.start(appId);
+                    }
+                }
             } else if (event === "launch") {
                 // FIXME: Remove once "splash" event works.
                 if (views.spinner && !views.spinner.isOpen && appId != null) {
